### ISSUE TITLE
[🐛 Bug]: Since v8 --spec only works with full path to spec file

### ISSUE THREAD

**Issue by synedra-mpe** [webdriverio/webdriverio#9495]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

16

### Mode

WDIO Testrunner

### What happened?

Before v8 it was possible to use the `--spec` option with only mentioning the spec file. wdio then figured out, which spec file to load, depending on den given specs glob pattern config. This was very convenient to quickly limit test execution to just one spec file and not the whole suite.

### What is your expected behavior?

Entering, for configured specs glob pattern `test/itests/js/specs/mainProduct/**/*Test.js` 

`wdio run --spec LoginTest.js` should be the same as eg.

`wdio run --spec test/itests/js/specs/mainProduct/LoginTest.js`

### How to reproduce the bug.

Make use of the `--spec` cli option in a wdio project without giving the full path to the spec js file.

### Relevant log output

```
2022-12-30T16:00:31.536Z WARN @wdio/config:ConfigParser: pattern test/itests/js/specs/mainProduct/**/*Test.js did not match any file
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!



Any contributions that resolves the bug are highly appreciated. Please take a look into our [contribution guidelines](https://webdriver.io/docs/contribute) and let us know if you have any questions. Cheers!

**Comment by ifero** [comment]
I have the same issue. Wanted to bump wdio version to v8+
but after the upgrade it doesn't find any spec anymore with the same error as @synedra-mpe
In v7 I just had to run the test-suite without specifying any `--spec` details
In v8 it fails

```
WARN @wdio/config:ConfigParser: pattern __tests__/e2e/features/**/*.features did not match any file
```

**Comment by ifero** [comment]
I believe it has something to do with the changes in [THIS LINE](https://github.com/webdriverio/webdriverio/blob/e053d0cf8e0da651237a45f40ce19fe61dbc031e/packages/wdio-config/src/lib/ConfigParser.ts#L399)
which now includes `rootDir` too as parameter

**Comment by ebonom-dazn** [comment]
I am experiencing the same issue after I updated to v8. Prior to v* upgrade all my spec files were found. Team webdriverio, please is there a solution to this issue?
![Screenshot 2023-02-14 at 13 58 52](https://user-images.githubusercontent.com/110390632/218759429-6cc87c79-4de5-4212-b601-9ed1dac01035.png)

**Comment by christian-bromann** [comment]
> Team webdriverio, please is there a solution to this issue?

Team webdriverio also includes you so any contributions are appreciated, thanks!

**Comment by leodsc** [comment]
Hi @christian-bromann, I am currently working on this. It seems that it really is a problem with [rootDir from the ConfigParser.ts](https://github.com/webdriverio/webdriverio/blob/31965e2a8cae9221cb69a6fc891a4de7d2f61cd3/packages/wdio-config/src/lib/ConfigParser.ts#L56). I was able to make it work correctly, but I am fixing when using wildcards for the spec files.

**Comment by christian-bromann** [comment]
@leodsc interesting, if you want to pair program on this feel free to book a slot during the open office hours

**Comment by leodsc** [comment]
@christian-bromann sure, I will schedule a meeting in calendly to make sure that what I did makes sense, thank you!

**Comment by vitalyiegorov** [comment]
@christian-bromann Any news? This is very uncomfortable to use full path

**Comment by christian-bromann** [comment]
@vitalyiegorov as you can see @leodsc has taken a stab at this. You will see updates here once they are available.

**Comment by dons20** [comment]
Is there any workaround we can use in the meantime while this is being worked on? Also, it would be useful to have a draft PR opened if there are any changes in progress. If not, maybe someone else would be able to take over the effort.

**Comment by christian-bromann** [comment]
No one is working on this, if you are interested in a fix, get involved and contribute time into the project, thank you!

**Comment by andrew-tompkins** [comment]
So I've been looking into this issue a little bit. And I have something working on my local machine. It will work if you pass any of these with the spec argument, this is assuming the filename for your test is `login.test.js`:
* `--spec login`
* `--spec login.test.js`
* `--spec directory/path/to/test/login.test.js`

But it's not very clean as it goes through every directory in your current working directory looking for a match. 

Should I just create a PR and let the review go from there? My fix is updating `makeRelativeToCWD` in the @wdio\config\build\utils.js file.

**Comment by christian-bromann** [comment]
@andrew-tompkins thanks for taking a stab at it. I am wondering why you have to search through the directory again? The spec parameter should only filter based on what the `ConfigParser` finds based on the `specs` property and the glob package. Also there is already an implementation for this in the `ConfigParser` , for some reasons it just doesn't work no more.

**Comment by andrew-tompkins** [comment]
@christian-bromann The reason I ended up changing the `makeRelativeToCWD` function was because when the ConfigParser is initialized, it's calling this function. And when a single --spec is set this function just returns the current working directly and the spec name. Ex. "/my/currrent/direct/login" (even though this is not the correct location), and I found that if this was set to the correct path for the spec then it was working. I did wonder though if this was the correct spot for a fix. So I'll have a look again and see if I can figure out a better way.

Thanks for the feedback.

**Comment by andrew-tompkins** [comment]
I was able to create a PR for this potential fix. (Noted above.)

**Comment by christian-bromann** [comment]
Fantastic @andrew-tompkins , thank you so much, let me close this and let's continue our convo in the PR. Thank you for taking a stab at this!

**Comment by bhanuagarwal73** [comment]
Team,

What's the fix ??

I am still getting the error. Can someone please help me out?
![image](https://github.com/webdriverio/webdriverio/assets/38572421/dfbea624-755f-4200-87f2-45788ad84e75)

**Comment by christian-bromann** [comment]
> What's the fix ??

As I mentioned numerous times, if you like something to get shipped, get involved!

**Comment by himalbandara84** [comment]
> Team,
> 
> What's the fix ??
> 
> I am still getting the error. Can someone please help me out? ![image](https://user-images.githubusercontent.com/38572421/253017904-dfbea624-755f-4200-87f2-45788ad84e75.png)

hey im had same problem what i did i just add two dots like this ../test/specs/login-spec.ts now its working

**Comment by mind-aman** [comment]
Hi Folks, I was facing the same issue but now I got it working with a small change. Not sure, if it will work for all but worth giving it a try.

```
WDIO v8
Node 18
Chrome latest
```

So in my local.conf.js file I was earlier accessing the file as below with WDIO v7 and node 14+ :

```
specs: ['features/**/*.feature'],
  suites: {
    myAllTests: ['features/**/*.feature']
}
```

However with WDIO v8 i simply added ../ to reach to correct spec or suite:

Eg:
```
specs: ['../features/**/*.feature'],
  suites: {
    myAllTests: ['../features/**/*.feature']
}
```


Now I can run any spec or suite using the command which I usually used -

`npx wdio wdio.local.conf.js --suite myAllTests`

`npx wdio wdio.local.conf.js --spec ./feature/**/*.feature`

Hope this works for others as well.

**Comment by christian-bromann** [comment]
@mind-aman please not that one of the breaking changes in v8 was that file paths are resolved from the WDIO config location rather than the current working directory.

**Comment by tech-dm-klymenko** [comment]
I'm going to start working on this issue

**Comment by NaZaRKIN123** [comment]
> Hi Folks, I was facing the same issue but now I got it working with a small change. Not sure, if it will work for all but worth giving it a try.
> 
> ```
> WDIO v8
> Node 18
> Chrome latest
> ```
> 
> So in my local.conf.js file I was earlier accessing the file as below with WDIO v7 and node 14+ :
> 
> ```
> specs: ['features/**/*.feature'],
>   suites: {
>     myAllTests: ['features/**/*.feature']
> }
> ```
> 
> However with WDIO v8 i simply added ../ to reach to correct spec or suite:
> 
> Eg:
> 
> ```
> specs: ['../features/**/*.feature'],
>   suites: {
>     myAllTests: ['../features/**/*.feature']
> }
> ```
> 
> Now I can run any spec or suite using the command which I usually used -
> 
> `npx wdio wdio.local.conf.js --suite myAllTests`
> 
> `npx wdio wdio.local.conf.js --spec ./feature/**/*.feature`
> 
> Hope this works for others as well.

It works on Mac, but not on Linux. Which is super strange...

**Comment by NaZaRKIN123** [comment]
```
wdiotests/
├── .husky/
├── .vscode/
├── reports/
├── config/
└── src/tests/
```
With the above folder structure, this suite works on Mac but not Linux. (it was working on WDIO 7)
```
config.suites = {
  web: ['../src/tests/*.feature'],
}
```
The error:
<img width="723" alt="image" src="https://github.com/webdriverio/webdriverio/assets/19157402/ec1248c7-c16b-45ac-bf1c-e8bf878c7b94">

**Comment by tech-dm-klymenko** [comment]
@christian-bromann 
Sure, I'll take a look at it, but only on Windows. Unfortunately, I don’t have the opportunity to check this on Linux now

**Comment by christian-bromann** [comment]
It seems @NaZaRKIN123 bug is unrelated to this feature.

**Comment by json-warner** [comment]
the spec config documentation for v8 - https://webdriver.io/docs/configurationfile/ - still states the working directory is based on location of package.json :
```
    // If you are calling `wdio` from an NPM script (see https://docs.npmjs.com/cli/run-script),
    // then the current working directory is where your `package.json` resides, so `wdio`
    // will be called from there.
```

does updating this require a new issue be opened?

**Comment by christian-bromann** [comment]
@json-warner thanks for pointing this out, I updated the documentation.

Closing as it got fixed in #11771

**Comment by ZehavitSaban** [comment]
> @christian-bromann Sure, I'll take a look at it, but only on Windows. Unfortunately, I don’t have the opportunity to check this on Linux now



> It seems @NaZaRKIN123 bug is unrelated to this feature.

@christian-bromann @tech-dm-klymenko 
I have the same issue in linux , any update on this?

**Comment by christian-bromann** [comment]
@ZehavitSaban can you explain what issue your are seeing?

**Comment by ZehavitSaban** [comment]
> @ZehavitSaban can you explain what issue your are seeing?
Sure, thanks for the help!

We are running wdio on AWS Lambda Node 20, using the devtools protocol (via devtools-service and puppeteer).

First, we build and deploy lambda with the following dependencies. So, when we invoke the Lambda, it already comes with these dependencies pre-installed on the Lambda.

  ```
"dependencies": {
    "@aws-sdk/client-s3": "3.465.0",
    "@sparticuz/chromium": "119.0.0",
    "@types/jest": "29.5.11",
    "@wdio/cli": "8.32.2",
    "@wdio/devtools-service": "8.32.2",
    "@wdio/local-runner": "8.32.2",
    "@wdio/mocha-framework": "8.32.2",
    "@wdio/spec-reporter": "8.32.2",
    "axios": "1.6.2",
    "chai": "4.3.10",
    "chalk": "4.1.2",
    "dotenv": "16.3.1",
    "lodash": "4.17.21",
    "moment": "2.29.4",
    "node-html-parser": "6.1.11",
    "puppeteer": "21.6.0",
    "puppeteer-core": "21.6.0",
    "stream": "0.0.2",
    "ts-node": "10.9.2",
    "ts-paths-esm-loader": "1.4.3",
    "tsconfig-paths": "4.2.0",
    "typescript": "5.3.3",
    "unzipper": "0.10.14",
    "uuid": "9.0.1"
  },
  "devDependencies": {
    "@dudadev/prettier-config": "1.0.1",
    "prettier": "3.1.0"
  }
```
  
Inside the lambda, we use S3 to get the company's internal automation framework and the test code we want to run. Then, we build the test env inside the lambda, create local configurations, wire things up, wire everything with puppeteer , and start the test.

Basically, with WDIO v7 it works fine. We can run tests inside the AWS Lambda and then integrate it with our CI/CD.
But, with WDIO v8 it doesn't work as you can see here:

[LAMBDA_DEBUG] Running command: [cd /tmp/wdio/ && timeout 125s node main.mjs]
[LAMBDA_DEBUG] See test code output in the next log:
Chrome path = function () { [native code] }
running spec global-buttons-edit-flex-test.js against https://my-test-master.dud4.co
Failed in wdio.run() command
Error: spec file(s) global-buttons-edit-flex-test.js not found
    at ConfigParser.setFilePathToFilterOptions (file:///private/tmp/wdio/node_modules/@wdio/config/build/node/ConfigParser.js:322:19)
    at ConfigParser.merge (file:///private/tmp/wdio/node_modules/@wdio/config/build/node/ConfigParser.js:168:39)
    at ConfigParser.initialize (file:///private/tmp/wdio/node_modules/@wdio/config/build/node/ConfigParser.js:50:14)
    at async Launcher.run (file:///private/tmp/wdio/node_modules/@wdio/cli/build/launcher.js:41:9)
    at async main (file:///private/tmp/wdio/main.mjs:27:32)
Saving lambdaSpecResultJson to file /tmp/wdio/reports/lambdaSpecResultJson.json
Successfully saved the file.

**Comment by christian-bromann** [comment]
> Error: spec file(s) global-buttons-edit-flex-test.js not found

How to you define `specs` in your `wdio.conf.ts` and how to you call WebdriverIO? Also, what is the absolute path of the WebdriverIO config and the `global-buttons-edit-flex-test.js` file?

**Comment by ZehavitSaban** [comment]
> How do you define `specs` in your `wdio.conf.ts` and how do you call WebdriverIO? Also, what is the absolute path of the WebdriverIO config and the `global-buttons-edit-flex-test.js` file?

How do you define `specs` in your `wdio.conf.ts`?
Answer : I don't know if this is relevant, but I changed the extension of the file from wdio.conf.ts to wdio.conf.mjs
 ```
   specs: ['./out/main/specs/**/*',
        './out/wdio-automation-infra-tests/**/*'],
```

How do you call WebdriverIO?
Answer :
 ```
   const wdio = new Launcher('./wdio.conf.mjs', {
        spec: [process.env.SPEC_FILE],
    });

    try {
        const specStatusCode = await wdio.run();
    } catch (e) {
        console.log('Failed in wdio.run() command');
        console.log(e);
    } 
```

 What is the absolute path of the WebdriverIO config and the `global-buttons-edit-flex-test.js` file?
 Answer :
**WebdriverIO config in aws lambda** - duda-lambdas/wdioSandbox/wdioSandbox/wdio.conf.mjs
 Inside the lambda, we use S3 to get the company's internal automation framework and the test code we want to run, so the `global-buttons-edit-flex-test.js` file is there : duda-wdio-js/src/main/specs/global-design/global-buttons/global-buttons-edit-flex-test.ts
 
When I run the test locally without using lambda(in duda-wdio-js project) everything works.
This is how I define specs there :
    specs: ['./src/main/specs/**/*', './src/wdio-automation-infra-tests/**/*'],
   **WebdriverIO config in duda-wdio-js project ** - duda-wdio-js/wdio.conf.ts

**Comment by ZehavitSaban** [comment]
Any update ? @christian-bromann

**Comment by christian-bromann** [comment]
If your config file is located in:

```
/uda-lambdas/wdioSandbox/wdioSandbox/wdio.conf.mjs
```

with the specs defined with

```ts
specs: [
    './out/main/specs/**/*',
    './out/wdio-automation-infra-tests/**/*'
]
```

Then the `global-buttons-edit-flex-test.js` needs to be located somewhere in

- `/uda-lambdas/wdioSandbox/wdioSandbox/out/main/specs/**/*`
- `/uda-lambdas/wdioSandbox/wdioSandbox/out/wdio-automation-infra-tests/**/*`

Can you confirm?

**Comment by ZehavitSaban** [comment]
> If your config file is located in:
> 
> ```
> /uda-lambdas/wdioSandbox/wdioSandbox/wdio.conf.mjs
> ```
> 
> with the specs defined with
> 
> ```ts
> specs: [
>     './out/main/specs/**/*',
>     './out/wdio-automation-infra-tests/**/*'
> ]
> ```
> 
> Then the `global-buttons-edit-flex-test.js` needs to be located somewhere in
> 
> * `/uda-lambdas/wdioSandbox/wdioSandbox/out/main/specs/**/*`
> * `/uda-lambdas/wdioSandbox/wdioSandbox/out/wdio-automation-infra-tests/**/*`
> 
> Can you confirm?

@christian-bromann
No, it doesn't work that way.
We don't have the specs at all in the duda-lambdas project.
It's in another project.
Inside the lambda, we use S3 to get the company's internal automation framework(the other project) and the test code we want to run. Then, we build the test env inside the lambda, create local configurations, wire things up, wire everything with puppeteer , and start the test.
Basically, with WDIO v7 it works fine. 

In S3 (to get the company's internal automation framework) we create a zip that contains this code (the second project) and we have this script that we run:
`sh script: "cp -r ./src ./out"`
Then all the specs should appear and it should be fine (and we have it):
./out/main/specs/global-design/global-buttons/global-buttons-edit-flex-test.js
But it doesn't work on v8 :
```
Failed in wdio.run() command
Error: spec file(s) global-buttons-edit-flex-test.js not found
```

**Comment by christian-bromann** [comment]
> Basically, with WDIO v7 it works fine.

The only difference in v8 compared to v7 that can be related to this is:

> Furthermore did we fix the behavior of relative spec or exclude paths. Before v8 every path within specs, exclude or --spec was always seen relative from the users working directory.

**Comment by ZehavitSaban** [comment]
> > Basically, with WDIO v7 it works fine.
> 
> The only difference in v8 compared to v7 that can be related to this is:
> 
> > Furthermore did we fix the behavior of relative spec or exclude paths. Before v8 every path within specs, exclude or --spec was always seen relative from the users working directory. This behavior was confusing especially when the wdio.conf.js was not within the root of you project. This got fixed now so that specs and exclude path will be always seen as relative to the config file and --spec arguments, relative from the working directory.

@christian-bromann True, it is related to this change.
The question is what can I do in my code to make it work in v8?

**Comment by christian-bromann** [comment]
> The question is what can I do in my code to make it work in v8?

Please update your path defined in `specs` and make it relative to the config file instead of your working directory.

**Comment by ZehavitSaban** [comment]
> > The question is what can I do in my code to make it work in v8?
> 
> Please update your path defined in `specs` and make it relative to the config file instead of your working directory.

@christian-bromann 
You mean I should change this :
specs: [
    './out/main/specs/**/*',
    './out/wdio-automation-infra-tests/**/*'
]
to this:
specs: [
    './duda-lambdas/wdioSandbox/wdioSandbox/out/main/specs/**/*',
   ' ./duda-lambdas/wdioSandbox/wdioSandbox/out/wdio-automation-infra-tests/**/*'
]

Please confirm.

**Comment by ZehavitSaban** [comment]
> I can't confirm because I don't know where your config file and tests are located.

Hey @christian-bromann,
We don't have the specs at all in the duda-lambdas project.
**It's in another project.**
Inside the lambda, we use S3 to get the company's internal automation framework(the other project) and the test code we want to run. Then, we build the test env inside the lambda, create local configurations, wire things up, wire everything with puppeteer , and start the test.
Basically, with WDIO v7 it works fine.

In S3 (to get the company's internal automation framework) we create a zip that contains this code (the second project) and we have this script that we run:
`sh script: "cp -r ./src ./out"
`Then all the specs should appear and it should be fine (and we have it):
./out/main/specs/global-design/global-buttons/global-buttons-edit-flex-test.js
But it doesn't work on v8 :

```
Failed in wdio.run() command
Error: spec file(s) global-buttons-edit-flex-test.js not found
```

**Comment by christian-bromann** [comment]
> Basically, with WDIO v7 it works fine.

I can't help you but pointing out to what I've mentioned before. Verify that the specs defined in your `specs` path are correct and relative the config file.

**Comment by ZehavitSaban** [comment]
@christian-bromann Thanks!
